### PR TITLE
Passing environment to the container.

### PR DESCRIPTION
**Pull request by adarsh-zededa**
Here is a document explaining the changes made in this PR:
https://docs.google.com/document/d/1ii303aLskeEvc36ZNUNbMCJteXtPcpoZAq_eld8rPVg/edit?usp=sharing

### CODE REVIEW

**Comment by deitch** [comment]
I have some concerns about the changes here. They can be split into the more fundamental, and then the more tactical.

From a fundamental perspective, should we be putting effort into rkt? The tool is dead, we already ripped out `rkt fetch` to be replaced by something saner, and we are doing work on `runx` to get rid of `rkt` entirely. I would be hesitant to put any real effort in. I get that we might need some additional support (like env vars), but can we confirm that this is critical for now?

From a tactical perspective, if we _really_ need to do this, we should not add new `pkg/` like `actool` or `docker2aci`. 

* We already have `docker2aci` as a library and can do anything it does inside `domainmgr`, which we do.
* We have reached the point where the entire usage of `rkt` is ring-fenced into `domainmgr` and appears nowhere else. We should keep it that way, or it will be _much_ harder to rip out afterwards.


More importantly, if all we are doing is passing environment variables, we don't need all of this. We just can add them to `rkt run --set-env=NAME1=value1 --set-env=NAME2=value2 ... ` and be done with it. It would be a far simpler PR.

In short, unless there is a really strong reason to do this way that we missed, can we simplify it down to just changing the `rkt run` command-line?

**Comment by rvs** [comment]
Hey @adarsh-zededa a few point:
  * First of all, I don't have access to the Google doc you linked to
  * Your PR seems to need rebasing, etc. since it is obviously bringing back changes that were removed in previous commits that landed in master. E.g.
```
TpmEnabledFile = types.IdentityDirname + "/tpm_in_use"
```
  * There's absolutely no need for a separate package for the use case you have

And finally, I absolutely agree with @deitch 

> if all we are doing is passing environment variables, we don't need all of this. We just can add them to rkt run --set-env=NAME1=value1 --set-env=NAME2=value2 ... and be done with it. It would be a far simpler PR.

So I'd like this PR to be simplified to the absolute bare minimum.

**Comment by rvs** [comment]
Hey @adarsh-zededa -- can we close this now?

**Comment by rvs** [comment]
Ping @adarsh-zededa ? I think this needs to be closed now, no?

**Comment by adarsh-zededa** [comment]
Closing this PR as an alternative solution (https://github.com/lf-edge/eve/pull/548) was given.
